### ISSUE TITLE
Can't override `skip` option in local config: `svType != tvType`

### ISSUE THREAD

**Issue by Envek** [evilmartians/lefthook#200]
Seems like there is some validation of config types exists that prevents overriding explicitly set `skip` option in local config.

Given following `lefthook.yml`:
```yml
post-checkout:
  piped: true
  scripts:
    01-bundle-checkinstall:
      skip: true
      tags: backend
    02-db-migrate:
      skip: true
      tags: backend
    03-crystalball-update:
      skip: true
      tags: rspec backend
```

And following `lefthook-local.yml`:
```yml
post-checkout:
  piped: true
  scripts:
    01-bundle-checkinstall:
      skip: false
    02-db-migrate:
      skip: rebase
    03-crystalball-update:
      skip:
        - rebase
        - merge
```

Checkout to another branch (without merge or rebase in progress) produces following output:

```
$ git checkout master

ERROR 2021/05/22 12:16:28 svType != tvType; key=skip, st=string, tt=bool, sv=rebase, tv=true
ERROR 2021/05/22 12:16:28 svType != tvType; key=skip, st=[]interface {}, tt=bool, sv=[rebase merge], tv=true
Lefthook v0.7.5
RUNNING HOOKS GROUP: post-checkout

  EXECUTE > 01-bundle-checkinstall
The Gemfile's dependencies are satisfied

02-db-migrate (SKIP BY SETTINGS)

03-crystalball-update (SKIP BY SETTINGS)

SUMMARY: (done in 0.36 seconds)
```

**Expected result**: all steps are run

**Actual result**: steps 2 and 3 are skipped, type errors are printed to console

**Additional details**: Upstream issue: https://github.com/spf13/viper/issues/1142

_Originally posted by @Envek in https://github.com/evilmartians/lefthook/pull/173#pullrequestreview-666167804_

**Comment by novitckas** [comment]
Any news on this issue?

Viper fixed issue on their side in v1.11.0. Seems like the only thing left to do is to update viper version in lefthook.

**Comment by mrexox** [comment]
I have tested this on #308, and it seems to fix the issue.
